tests/functional/builtins/codegen/test_blobhash.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+
+from vyper import compiler
+from vyper.exceptions import EvmVersionException
+
+valid_list = [
+    """
+@external
+@view
+def foo() -> bytes32:
+    return blobhash(0)
+    """,
+    """
+@external
+@view
+def foo() -> bytes32:
+    a: bytes32 = 0x0000000000000000000000000000000000000000000000000000000000000005
+    a = blobhash(2)
+    return a
+    """,
+    """
+@external
+@view
+def foo() -> bytes32:
+    a: bytes32 = blobhash(5)
+    return a
+    """,
+    """
+@external
+@view
+def foo() -> bytes32:
+    a: bytes32 = blobhash(1337)
+    assert a == empty(bytes32)
+    return a
+    """,
+]
+
+
+@pytest.mark.requires_evm_version("cancun")
+@pytest.mark.parametrize("good_code", valid_list)
+def test_blobhash_success(good_code):
+    assert compiler.compile_code(good_code) is not None
+    assembly = compiler.compile_code(good_code, output_formats=["asm"])["asm"].split(" ")
+    assert "BLOBHASH" in assembly
+
+
+@pytest.mark.requires_evm_version("shanghai")
```

Comment:
i don't think this is necessary, it's already covered by the above test

Reply:
hmm, we should always write positive and negative tests IMO.

Reply:
check what `requires_evm_version` does, it already ensures that EvmVersionException is thrown if version<cancun.

Reply:
removed this test via ab08210aa9edb00c2841b6f9a67a726fae4dffba